### PR TITLE
Fix the wrong status returned from `RunPreFilterPlugins`

### PR DESCRIPTION
**Pull request by chendave**
event msg will be impacted due to the wrong status is returned.
e.g.
```
Warning  FailedScheduling ... running PreFilter plugin "PodTopologySpread": %!!(MISSING)w(<nil>)
```
/kind bug

Signed-off-by: Dave Chen <dave.chen@arm.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig scheduling

### CODE REVIEW

File: pkg/scheduler/framework/runtime/framework_test.go [hunk 1]
```diff
@@ -1240,6 +1240,36 @@ func TestPreFilterPlugins(t *testing.T) {
...
+	}
+	r := make(Registry)
+	r.Register(preFilterPluginName,
+		func(_ runtime.Object, fh framework.Handle) (framework.Plugin, error) {
+			return preFilter, nil
+		})
+
+	plugins := &config.Plugins{PreFilter: config.PluginSet{Enabled: []config.Plugin{{Name: preFilterPluginName}}}}
+
+	t.Run("TestPreFilterPluginStatus", func(t *testing.T) {
```
Comment (sanposhiho) [inline pkg/scheduler/framework/runtime/framework_test.go:1256]: I'd say you don't need to make it the subtest.

**Review by sanposhiho — COMMENTED** [review]
nice catch!

File: pkg/scheduler/framework/runtime/framework_test.go (hunk 1 above)
Comment (chendave) [reply pkg/scheduler/framework/runtime/framework_test.go:1256]: I expect this testcase to expand a little bit to cover more scenario of `RunPreFilterPlugins`, but for now, yes, it's only used to test the fixing of the bug.

Comment (sanposhiho) [reply pkg/scheduler/framework/runtime/framework_test.go]: If considering the future expansion, we can implement it as table-driven tests with only one table for now.
I left it up to you to decide; implement as table-driven tests or as a single test as it is now. Either way seems fine to me.

**Review by sanposhiho — COMMENTED** [review]
/lgtm
/retest

**Comment by ahg-g** [comment]
/approve

Thanks

**Pull request merged**